kolibri/core/tasks/api.py

```diff
@@ -238,26 +53,34 @@ def validate_create_req_data(self, request):
         else:
             request_data_list = [request.data]
 
+        validated_jobs = []
+
         for request_data in request_data_list:
-            if "task" not in request_data:
-                raise serializers.ValidationError("The 'task' field is required.")
-            if not isinstance(request_data["task"], string_types):
-                raise serializers.ValidationError("The 'task' value must be a string.")
+            # Make sure the task is registered
+            registered_task = TaskRegistry.validate_task(request_data.get("type"))
 
-            funcstr = request_data.get("task")
+            job = registered_task.validate_job_data(request.user, request_data)
 
-            # Make sure the task is registered
-            try:
-                registered_job = JobRegistry.REGISTERED_JOBS[funcstr]
-            except KeyError:
-                raise serializers.ValidationError(
-                    "'{funcstr}' is not registered.".format(funcstr=funcstr)
-                )
+            registered_task.check_job_permissions(request.user, job, self)
 
-            # Check permissions
-            self.check_registered_job_permissions(request, registered_job)
+            validated_jobs.append((registered_task, job))
 
-        return request_data_list
+        return validated_jobs
+
+    def _job_to_response(self, job):
+        output = {
+            "status": job.state,
+            "type": job.func,
+            "exception": job.exception,
+            "traceback": job.traceback,
+            "percentage": job.percentage_progress,
+            "id": job.job_id,
+            "cancellable": job.cancellable,
+            "clearable": job.state in [State.FAILED, State.CANCELED, State.COMPLETED],
+            "facility_id": job.facility_id,
+            "extra_metadata": job.extra_metadata,
```

Comment:
`extra_metadata` has been removed in the above `ExportLogCSVValidator`, but here I see it's still available to be used, thus keeping it in the `ExportLogCSVValidator` would save frontend work. Is there any intention to deprecate it or what's the reason to have it available but not using it?

Reply:
Still intended to be used, I just migrated all of the previous `extra_metadata` in that specific task to either always be added, or be redundant with other data already on the job object.